jib-core/src/main/java/com/google/cloud/tools/jib/blob/Blobs.java

```diff
@@ -42,7 +48,7 @@ public static Blob from(BlobWriter writer) {
   }
 
   /**
-   * Writes the BLOB to a string.
+   * Writes the BLOB to a string with UTF-8 decoding.
```

Comment:
This is where the carry-files-as-strings breaks down: although any String can be encoded as a set of bytes in UTF-8, not all bytes are valid UTF-8 strings.

If you want to go down this route, I'd much prefer `Blob encodeStringBlob()` and `decodeStringBlob()`